paramiko/rsakey.py

```diff
@@ -204,3 +212,21 @@ def _decode_key(self, data):
             self._got_bad_key_format_id(pkformat)
         assert isinstance(key, rsa.RSAPrivateKey)
         self.key = key
+
+
+class RSASHA256Key(RSAKey):
+    """
+    A special RSAKey that uses SHA-256 digest for sign/verify.
+    """
+
+    _signature_digest_algorithm = hashes.SHA256
+    _signature_algorithm_name = "rsa-sha2-256"
+
+
+class RSASHA512Key(RSAKey):
+    """
+    A special RSAKey that uses SHA-256 digest for sign/verify.
```

Comment:
Is it a typo here? Should it be `SHA-512`?

Reply:
Yes, I think so.